core/src/services/pilot.rs

```diff
@@ -0,0 +1,159 @@
+use anyhow::{Result};
```

Comment:
Thanks for moving this into the core!

Reply:
I believe this is actually from @qtpeters branch. I rebased from Main and then his branch so he should get credit for doing this!